code/API_definitions/Application_Endpoint_Discovery_API.yaml

```diff
@@ -0,0 +1,596 @@
+          description: |
+            When the API Consumer includes the "X-Correlator" header in the
+            request, the API provider must include it in the response with
+            the same value that was used in the request. Otherwise, it is
+            optional to include the "X-Correlator" header in the response with
+             any valid value. Recommendation is to use UUID for values.
+          schema:
+            type: string
+            format: uuid
+      responses:
+        "200":
+          description: |
+            Successful response, returning the closest App Instance Endpoint
+            Zone to the user device identified in the request header.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/AccessEndpoint"
+        "400":
+          $ref: "#/components/responses/400BadRequest"
+        "401":
+          $ref: "#/components/responses/401Unauthorized"
+        "403":
+          $ref: "#/components/responses/403Forbidden"
+        "404":
+          $ref: "#/components/responses/404NotFound"
+        "405":
+          $ref: "#/components/responses/405MethodNotAllowed"
+        "406":
+          $ref: "#/components/responses/406Unacceptable"
+        "429":
+          $ref: "#/components/responses/429TooManyRequests"
+        "500":
+          $ref: "#/components/responses/500InternalServerError"
+        "502":
+          $ref: "#/components/responses/502BadGateway"
+        "503":
+          $ref: "#/components/responses/503ServiceUnavailable"
+        "504":
+          $ref: "#/components/responses/504GatewayTimeout"
+      tags:
+        - Discovery
+      summary: |
+        Returns the Endpoint of the closest App Instance to user device
+        identified in the request.
+      description: |
+        On receiving this request, the network will return the name
+        of the App Instance with the shortest network path to the end user
+        device identified in the request.
+components:
+  securitySchemes:
+    openId:
+      description: OpenID Provider Configuration Information.
+      type: openIdConnect
+      openIdConnectUrl: .well-known/openid-configuration
+  headers:
+    X-Correlator:
+      description: |
+       When the API Consumer includes the "X-Correlator" header in the request,
+        the API provider must include it in the response with the same value t
+        hat was used in the request. Otherwise, it is optional to include the
+        "X-Correlator" header in the response with any valid value.
+        Recommendation is to use UUID for values.
+      required: false
+      schema:
+        type: string
+        format: uuid
+  schemas:
+    AppId:
+      type: string
+      format: uuid
+      description: A globally unique identifier associated with the application.
+        OP generates this identifier when the application is submitted over NBI.
+    AccessEndpoint:
+      type: object
```

Comment:
I think for AccessEndpoint we need more discussion in terms of the data model. The reason is what should we expect in terms of the number of exposure endpoints of an application towards the application clients. May be an application is composed of multiple components and offers one or more endpoints to its clients for different purposes. In that case data model should consider those possibilities and accordingly consider them in defining the  AccessEndpoint structure e.g. one or more ports, fqdns etc as an example.

Reply:
Why `AccessEndpoint` and not `ApplicationServerEndpoint`?

Reply:
OK. Agree with AppInstanceEndpoint and as in the comment above in line 134, AppServerEndpoint is more specific and we are refering mostly the application in general